chroma-common/blockdevices/blockdevice_zfs.py

```diff
@@ -241,33 +241,39 @@ def filesystem_info(self):
 
         :return: message describing zfs type and datasets found, None otherwise
         """
-        self._assert_zpool('filesystem_info')
+        try:
+            self._assert_zpool('filesystem_info')
+        except NotZpoolException:
+            blockdevice = BlockDevice(self._supported_device_types[0], self._device_path.split('/')[0])
 
-        with ZfsDevice(self._device_path, False):
-            try:
-                device_names = shell.Shell.try_run(['zfs', 'list', '-H', '-o', 'name', '-r', self._device_path]).split('\n')
+            return blockdevice.filesystem_info
+        else:
+            with ZfsDevice(self._device_path, False):
+                try:
+                    device_names = shell.Shell.try_run(['zfs', 'list', '-H', '-o', 'name', '-r', self._device_path]).split('\n')
 
-                datasets = [line.split('/', 1)[1] for line in device_names if '/' in line]
+                    datasets = [line.split('/', 1)[1] for line in device_names if '/' in line]
 
-                if datasets:
-                    return "Dataset%s '%s' found on zpool '%s'" % ('s' if (len(datasets) > 1) else '',
-                                                                   ','.join(datasets),
-                                                                   self._device_path)
-                return None
-            except OSError:                             # zfs not found
-                return "Unable to execute commands, check zfs is installed."
-            except shell.Shell.CommandExecutionError as e:    # no zpool 'self._device_path' found
-                return str(e)
+                    if datasets:
+                        return "Dataset%s '%s' found on zpool '%s'" % ('s' if (len(datasets) > 1) else '',
+                                                                       ','.join(datasets),
+                                                                       self._device_path)
+                    return None
+                except OSError:                             # zfs not found
+                    return "Unable to execute commands, check zfs is installed."
+                except shell.Shell.CommandExecutionError as e:    # no zpool 'self._device_path' found
+                    return str(e)
```

Comment:
So, yes, we are checking that the ZFS structures (pools, datasets, etc.) are in place to provide a device which can be a Lustre target, but I don't see where we are actually reading that device (i.e. like we do with ldiskfs using `blkid`) to verify that it's contents are actually a Lustre ZFS target that has successfully been formatted with mkfs.lustre.

Probably a better way than `blkid` however, for both zfs and ldiskfs would be to run `tunefs.lustre` on the target after it's been formatted.  `tunefs.lustre`dat actually opens the device and reads the format from it and prints something such as:

```
# tunefs.lustre zfs_pool_scsi0QEMU_QEMU_HARDDISK_target5/testfs-OST0001
checking for existing Lustre data: found

   Read previous values:
Target:     testfs-OST0001
Index:      1
Lustre FS:  testfs
Mount type: zfs
Flags:      0x2
              (OST )
Persistent mount opts: 
Parameters: mgsnode=192.168.121.91@tcp

   Permanent disk data:
Target:     testfs-OST0001
Index:      1
Lustre FS:  testfs
Mount type: zfs
Flags:      0x2
              (OST )
Persistent mount opts: 
Parameters: mgsnode=192.168.121.91@tcp
```

I think that's the level of verification we need to get to the bottom of this bug.

Please also note my other comment about catching the "e2label: ..." on stderr and doing a `_fetch_help()` if it is hit.